### ISSUE TITLE
exist some error

### ISSUE THREAD

**Issue by ericyq** [JingXiaolun/TC-MGC#3]
the following files are missing:
in modules/modeling_tcmgc.py
self.logit_weight
self.frame_features_conversion_flag
self.radio
......
error:
until_module.py", line 188, in forward
    logpt = torch.diag(logpt)

**Comment by JingXiaolun** [comment]
Thanks for pointing out this mistake. We have updated the code in modeling_tcmgc.py, including adding self.logit_weight, removing self. frame_features_conversion_flag and self.radio.  The self. frame_features_conversion_flag and self.ratio are initially used to convert the text-conditioned frame dimension from num_words to num_frames. In our experiments, we find this operation harms the retrieval performance. As a result, we do not conduct this implementation in our accepted paper.

**Comment by ericyq** [comment]
thanks, successfully!
......
INFO:logger:  <<< visualize_dir: ./Visualize/tcmgc_didemo_vit32
INFO:logger:  <<< warmup_proportion: 0.1
INFO:logger:  <<< world_size: 4
INFO:logger:device: cuda:0 n_gpu: 4
INFO:logger:device: cuda:2 n_gpu: 4
INFO:logger:device: cuda:1 n_gpu: 4
INFO:logger:device: cuda:3 n_gpu: 4
WARNING:modules.modeling:Stage-One:True, Stage-Two:False
WARNING:modules.modeling:Test retrieval by loose type.
WARNING:modules.modeling:        embed_dim: 512
WARNING:modules.modeling:        image_resolution: 224
WARNING:modules.modeling:        vision_layers: 12
WARNING:modules.modeling:        vision_width: 768
WARNING:modules.modeling:        vision_patch_size: 32
WARNING:modules.modeling:        context_length: 77
WARNING:modules.modeling:        vocab_size: 49408
WARNING:modules.modeling:        transformer_width: 512
WARNING:modules.modeling:        transformer_heads: 8
WARNING:modules.modeling:        transformer_layers: 12
WARNING:modules.modeling:                linear_patch: 2d
WARNING:modules.modeling:        cut_top_layer: 0
WARNING:modules.modeling:        sim_header: seqTransf
INFO:logger:***** Running test *****
INFO:logger:  Num examples = 0
INFO:logger:  Batch size = 16
INFO:logger:  Num steps = 0
INFO:logger:***** Running val *****
INFO:logger:  Num examples = 0
INFO:logger:***** Running training *****
INFO:logger:  Num examples = 180000
INFO:logger:  Batch size = 16
INFO:logger:  Num steps = 56250
INFO:logger:Epoch: 1/5, Step: 20/11250, Lr: 0.000000000-0.000000356, Loss: 1.168544, Time/step: 0.588720
INFO:logger:Epoch: 1/5, Step: 40/11250, Lr: 0.000000001-0.000000711, Loss: 1.979594, Time/step: 0.452634
INFO:logger:Epoch: 1/5, Step: 60/11250, Lr: 0.000000001-0.000001067, Loss: 2.605160, Time/step: 0.446175
INFO:logger:Epoch: 1/5, Step: 80/11250, Lr: 0.000000001-0.000001422, Loss: 1.338918, Time/step: 0.458043
INFO:logger:Epoch: 1/5, Step: 100/11250, Lr: 0.000000002-0.000001778, Loss: 1.173338, Time/step: 0.451018
INFO:logger:Epoch: 1/5, Step: 120/11250, Lr: 0.000000002-0.000002133, Loss: 2.568925, Time/step: 0.438475
INFO:logger:Epoch: 1/5, Step: 140/11250, Lr: 0.000000002-0.000002489, Loss: 2.310739, Time/step: 0.457022
INFO:logger:Epoch: 1/5, Step: 160/11250, Lr: 0.000000003-0.000002844, Loss: 1.578658, Time/step: 0.451667
INFO:logger:Epoch: 1/5, Step: 180/11250, Lr: 0.000000003-0.000003200, Loss: 0.520543, Time/step: 0.455065
INFO:logger:Epoch: 1/5, Step: 200/11250, Lr: 0.000000004-0.000003556, Loss: 0.920081, Time/step: 0.438029
